# AI impact on students 

A synthetic student [database](https://www.kaggle.com/datasets/laveshjadon/ai-impact-on-students) is analyzed to study AI use by studens and its impact on academic performance.

Different relations are made to analyze use case and use intensity by student type.

## 1. Dataset charge and verification

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("./../../Datasets/ai_student_impact.csv")
df.head(10)

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,100001,Humanities,Senior,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
1,100002,Medical,Junior,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
2,100003,Business,Freshman,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,100004,Business,Senior,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
4,100005,STEM,Sophomore,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium
5,100006,STEM,Junior,3.449,6.50,Debugging/Troubleshooting,Beginner,1,False,14.19,4,Allowed_With_Citation,5,3.666,65.92,High
6,100007,STEM,Freshman,3.622,31.41,Summarizing_Reading,Advanced,5,True,13.11,8,Allowed_With_Citation,7,4.000,67.97,Medium
7,100008,Arts,Junior,2.746,5.33,Copywriting/Drafting,Intermediate,3,False,18.45,2,Actively_Encouraged,1,2.965,85.09,Medium
8,100009,Business,Sophomore,3.420,2.00,Debugging/Troubleshooting,Beginner,2,True,2.87,1,Strict_Ban,5,3.396,55.71,Medium
9,100010,Business,Sophomore,3.046,19.99,Debugging/Troubleshooting,Intermediate,2,True,12.49,3,Strict_Ban,8,2.978,87.18,High


The identifier is checked to find duplicates, and there is none 

In [3]:
df['Student_ID'].value_counts()

Student_ID
100001    1
100002    1
100003    1
100004    1
100005    1
         ..
149996    1
149997    1
149998    1
149999    1
150000    1
Name: count, Length: 50000, dtype: int64

Describe method is executed to see basic statistical indicators on the dataset.

In [4]:
indicators = df.describe()
indicators

,Student_ID,Pre_Semester_GPA,Weekly_GenAI_Hours,Tool_Diversity,Traditional_Study_Hours,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score
count,50000.000000,50000.000000,50000.000000,50000.00000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000
mean,125000.500000,3.146102,8.427752,2.80026,11.209271,3.505360,4.270760,3.349299,75.798125
std,14433.901067,0.478854,8.269490,1.18802,5.156426,1.820812,2.144066,0.495673,13.281626
min,100001.000000,1.183000,0.000000,1.00000,1.000000,1.000000,1.000000,1.000000,10.780000
25%,112500.750000,2.834000,2.390000,2.00000,7.560000,2.000000,3.000000,3.023750,66.820000
50%,125000.500000,3.210000,5.800000,3.00000,11.180000,3.000000,4.000000,3.421000,76.000000
75%,137500.250000,3.521000,11.720000,4.00000,14.710000,5.000000,6.000000,3.749000,85.190000
max,150000.000000,3.998000,40.000000,5.00000,35.860000,10.000000,10.000000,4.000000,100.000000


With this information it can be said that the quantity variables are measured with the following ranges

In [5]:
qualification_ranges = pd.DataFrame({   'Lower Bound': [1,1,1,1,1,1],
                                        'Upper Bound': [5,5,10,10,5,100]},
                                    index=['Pre_Semester_GPA', 'Tool_Diversity', 
                                           'Perceived_AI_Dependency', 
                                           'Anxiety_Level_During_Exams',
                                           'Post_Semester_GPA',
                                           'Skill_Retention_Score'])
qualification_ranges

,Lower Bound,Upper Bound
Pre_Semester_GPA,1,5
Tool_Diversity,1,5
Perceived_AI_Dependency,1,10
Anxiety_Level_During_Exams,1,10
Post_Semester_GPA,1,5
Skill_Retention_Score,1,100


## 2. Students count per category

### Students per study field

In [6]:
students_category = df[["Student_ID","Major_Category"]].groupby("Major_Category").count().reset_index()
students_category['Pecentage'] = round(students_category['Student_ID'] / 50000 * 100, 2)
students_category

,Major_Category,Student_ID,Pecentage
0,Arts,5933,11.87
1,Business,12538,25.08
2,Humanities,9994,19.99
3,Medical,6476,12.95
4,STEM,15059,30.12


### Students per year

This is the meaning of each category:

- Junior    => Third year of college
- Freshman  => First year of college
- Sophomore => Second year of college
- Senior    => Last year of college
- Graduate  => Degree completed

In [7]:
students_year = df[["Student_ID","Year_of_Study"]].groupby("Year_of_Study").count().reset_index()
students_year['Pecentage'] = round(students_year['Student_ID'] / 50000 * 100, 2)
students_year

,Year_of_Study,Student_ID,Pecentage
0,Freshman,11031,22.06
1,Graduate,7430,14.86
2,Junior,11045,22.09
3,Senior,10634,21.27
4,Sophomore,9860,19.72


### Students with AI subscription

In [8]:
students_subs = df[["Student_ID","Paid_Subscription"]].groupby("Paid_Subscription").count().reset_index()
students_subs['Pecentage'] = round(students_subs['Student_ID'] / 50000 * 100, 2)
students_subs

,Paid_Subscription,Student_ID,Pecentage
0,False,28846,57.69
1,True,21154,42.31


### Students per amount of AI tools used

In [9]:
students_AI_tool = df[["Student_ID","Tool_Diversity"]].groupby("Tool_Diversity").count().reset_index()
students_AI_tool['Pecentage'] = round(students_AI_tool['Student_ID'] / 50000 * 100, 2)
students_AI_tool

,Tool_Diversity,Student_ID,Pecentage
0,1,8141,16.28
1,2,11830,23.66
2,3,16926,33.85
3,4,8081,16.16
4,5,5022,10.04


### Students per usage proficiency

In [10]:
students_AI_skill = df[["Student_ID","Prompt_Engineering_Skill"]].groupby("Prompt_Engineering_Skill").count().reset_index()
students_AI_skill['Pecentage'] = round(students_AI_skill['Student_ID'] / 50000 * 100, 2)
students_AI_skill

,Prompt_Engineering_Skill,Student_ID,Pecentage
0,Advanced,13809,27.62
1,Beginner,18495,36.99
2,Intermediate,17696,35.39


### Students per institutional policy on AI usage

In [11]:
students_policy = df[["Student_ID","Institutional_Policy"]].groupby("Institutional_Policy").count().reset_index()
students_policy['Pecentage'] = round(students_policy['Student_ID'] / 50000 * 100, 2)
students_policy

,Institutional_Policy,Student_ID,Pecentage
0,Actively_Encouraged,14988,29.98
1,Allowed_With_Citation,25224,50.45
2,Strict_Ban,9788,19.58


## 3. Deep analysis

In this section students are grouped by study field to find relations between their study field and performance indicators, also between study field and AI indicators.

### Arts students

A new Dataframe is created to select only students of arts

In [12]:
arts_students = df.loc[df.Major_Category == "Arts"]
arts_students

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
7,100008,Arts,Junior,2.746,5.33,Copywriting/Drafting,Intermediate,3,False,18.45,2,Actively_Encouraged,1,2.965,85.09,Medium
12,100013,Arts,Senior,3.479,2.86,Copywriting/Drafting,Beginner,2,False,8.95,2,Actively_Encouraged,4,3.612,67.68,High
25,100026,Arts,Senior,3.308,2.38,Ideation,Advanced,4,True,11.33,2,Actively_Encouraged,2,4.000,81.85,High
35,100036,Arts,Sophomore,3.632,1.17,Direct_Answer_Generation,Intermediate,4,True,3.67,2,Strict_Ban,5,3.553,55.89,High
51,100052,Arts,Senior,2.433,11.16,Debugging/Troubleshooting,Beginner,2,True,12.82,6,Actively_Encouraged,5,2.903,69.95,High
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49933,149934,Arts,Senior,3.064,6.66,Copywriting/Drafting,Intermediate,1,False,1.00,2,Actively_Encouraged,2,3.092,83.48,Low
49943,149944,Arts,Senior,3.066,2.95,Copywriting/Drafting,Intermediate,2,False,4.81,3,Allowed_With_Citation,4,3.260,80.09,High
49963,149964,Arts,Junior,2.970,0.87,Summarizing_Reading,Intermediate,1,True,6.92,2,Allowed_With_Citation,3,3.331,83.66,Medium
49983,149984,Arts,Sophomore,3.245,1.42,Copywriting/Drafting,Beginner,3,False,10.30,3,Actively_Encouraged,4,3.385,85.93,Medium


A count of students per year of study is made to know how many there are in each year.

In [13]:
arts_students["Year_of_Study"].value_counts().sort_values()

Year_of_Study
Graduate      844
Sophomore    1182
Freshman     1282
Senior       1301
Junior       1324
Name: count, dtype: int64

A count of students that have an specific use of AI is made to know what is the most frequent use in each year of study.

In [14]:
arts_s_AI_use = arts_students.groupby(["Year_of_Study","Primary_Use_Case"]).count()
arts_s_AI_use = arts_s_AI_use[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
arts_s_AI_use = (
    arts_s_AI_use
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
arts_s_AI_use = arts_s_AI_use[["Year_of_Study","Primary_Use_Case","Student_ID"]]
arts_s_AI_use

,Year_of_Study,Primary_Use_Case,Student_ID
0,Freshman,Copywriting/Drafting,559
1,Freshman,Direct_Answer_Generation,196
2,Freshman,Ideation,195
3,Freshman,Debugging/Troubleshooting,172
4,Freshman,Summarizing_Reading,160
5,Graduate,Copywriting/Drafting,370
6,Graduate,Ideation,136
7,Graduate,Summarizing_Reading,125
8,Graduate,Debugging/Troubleshooting,112
9,Graduate,Direct_Answer_Generation,101


This table shows that Copywriting/Drafting is the main use in students of every year category, but the other use cases differ in the different categories. 

With this info it can be given the following statements: 

- Summarizing_Reading is the less important use for Freshman and Sophomore students.
- Debugging/Troubleshooting	represents the 3rd or 4th option among all types of students. 
- Direct_Answer_Generation is the less important use for Graduate and Junior students.
- Ideation seems to be important for Graduate students but not for the other categories.

#### **General stats**

Columns are sorted in a different order to have a clearer view of information, then describe method is called to look for basic statistical indicators.

In [15]:
arts_students_stats = arts_students[['Pre_Semester_GPA', 'Post_Semester_GPA',
                                       'Traditional_Study_Hours', 'Weekly_GenAI_Hours',
                                       'Tool_Diversity', 'Perceived_AI_Dependency',
                                       'Anxiety_Level_During_Exams', 'Skill_Retention_Score']]
arts_students_stats.describe()

,Pre_Semester_GPA,Post_Semester_GPA,Traditional_Study_Hours,Weekly_GenAI_Hours,Tool_Diversity,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Skill_Retention_Score
count,5933.000000,5933.000000,5933.000000,5933.000000,5933.000000,5933.000000,5933.000000,5933.000000
mean,3.146786,3.343688,11.352412,7.271702,2.762683,3.363391,4.174279,75.660947
std,0.479915,0.496339,5.144670,7.042327,1.191667,1.711516,2.098140,12.984435
min,1.396000,1.316000,1.000000,0.000000,1.000000,1.000000,1.000000,28.940000
25%,2.833000,3.014000,7.800000,2.070000,2.000000,2.000000,3.000000,66.830000
50%,3.216000,3.410000,11.320000,5.070000,3.000000,3.000000,4.000000,75.540000
75%,3.523000,3.743000,14.830000,10.160000,4.000000,4.000000,6.000000,84.820000
max,3.990000,4.000000,29.880000,36.000000,5.000000,10.000000,10.000000,100.000000


After analyzing Arts students stats it can be given these preliminary conclusions:

- Traditional study hours remains as the main study method for the students.
- The mean of tool diversity is 2.76, which means that students aren't interested on exploring different types of AI tools.
- Perception of AI dependency by students is low and anxiety level too, so this variables cannot prove a negative impact of AI usage.
- When GPA mean and max on pre and post semester is analyzed it can be concluded that there's no real impact of AI on student performance, that's because GPA post is only 0.2 points higher than GPA pre semester, and the maximum grade remains almost the same.

#### **Burnout risk**

The column Burnout_Risk_Level represents an assessment applied to students to measure risk of having burnout after completing one semester.

A count is made to know that is the most frequent risk type.

In [16]:
arts_burnout_risk = arts_students['Burnout_Risk_Level'].value_counts().reset_index()
arts_burnout_risk['Percentaje'] = round(arts_burnout_risk['count'] / 5933 * 100, 2)
arts_burnout_risk

,Burnout_Risk_Level,count,Percentaje
0,Medium,2573,43.37
1,Low,2014,33.95
2,High,1346,22.69


This means that almost half of students have medium risk level of having burnout at the end of semester, and High level only represents 22.68% which is good.

A count of students that have an specific risk level is made to know if there is a variation in the different years of study.

In [17]:
arts_s_year_bnt = arts_students.groupby(["Year_of_Study","Burnout_Risk_Level"]).count()
arts_s_year_bnt = arts_s_year_bnt[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
arts_s_year_bnt = (
    arts_s_year_bnt
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
arts_s_year_bnt = arts_s_year_bnt[["Year_of_Study","Burnout_Risk_Level","Student_ID"]]
arts_s_year_bnt

,Year_of_Study,Burnout_Risk_Level,Student_ID
0,Freshman,Low,540
1,Freshman,Medium,532
2,Freshman,High,210
3,Graduate,Medium,384
4,Graduate,High,287
5,Graduate,Low,173
6,Junior,Medium,585
7,Junior,Low,453
8,Junior,High,286
9,Senior,Medium,566


As seen in the table Freshman is the only Year_of_Study category that has a different main risk level.

### Business students

A new Dataframe is created to select only students of Business

In [18]:
business_students = df.loc[df.Major_Category == "Business"]
business_students

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
2,100003,Business,Freshman,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,100004,Business,Senior,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
8,100009,Business,Sophomore,3.420,2.00,Debugging/Troubleshooting,Beginner,2,True,2.87,1,Strict_Ban,5,3.396,55.71,Medium
9,100010,Business,Sophomore,3.046,19.99,Debugging/Troubleshooting,Intermediate,2,True,12.49,3,Strict_Ban,8,2.978,87.18,High
17,100018,Business,Senior,3.156,8.03,Direct_Answer_Generation,Advanced,4,False,10.70,4,Allowed_With_Citation,5,3.390,90.65,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49990,149991,Business,Senior,2.970,2.63,Summarizing_Reading,Intermediate,2,False,24.95,1,Allowed_With_Citation,1,3.238,84.87,Medium
49992,149993,Business,Graduate,2.829,2.59,Debugging/Troubleshooting,Advanced,3,True,7.54,1,Allowed_With_Citation,4,3.176,84.04,Low
49995,149996,Business,Senior,2.899,12.16,Copywriting/Drafting,Beginner,2,False,13.36,2,Allowed_With_Citation,2,3.584,66.16,High
49997,149998,Business,Senior,3.177,15.87,Summarizing_Reading,Advanced,5,True,3.92,4,Allowed_With_Citation,5,3.605,97.21,High


A count of students per year of study is made to know how many there are in each year.

In [19]:
business_students["Year_of_Study"].value_counts().sort_values()

Year_of_Study
Graduate     1845
Sophomore    2467
Senior       2550
Freshman     2831
Junior       2845
Name: count, dtype: int64

A count of students that have an specific use of AI is made to know what is the most frequent use in each year of study.

In [20]:
business_s_AI_use = business_students.groupby(["Year_of_Study","Primary_Use_Case"]).count()
business_s_AI_use = business_s_AI_use[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
business_s_AI_use = (
    business_s_AI_use
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
business_s_AI_use = business_s_AI_use[["Year_of_Study","Primary_Use_Case","Student_ID"]]
business_s_AI_use

,Year_of_Study,Primary_Use_Case,Student_ID
0,Freshman,Ideation,1320
1,Freshman,Summarizing_Reading,415
2,Freshman,Direct_Answer_Generation,371
3,Freshman,Debugging/Troubleshooting,363
4,Freshman,Copywriting/Drafting,362
5,Graduate,Ideation,904
6,Graduate,Direct_Answer_Generation,254
7,Graduate,Debugging/Troubleshooting,243
8,Graduate,Summarizing_Reading,226
9,Graduate,Copywriting/Drafting,218


This table shows that Ideation is the main use in students of every year category, but the other use cases differ in the different categories. 

With this info it can be given the following statements: 

- Copywriting/Drafting is the less important use for Freshman, Graduate and Sophomore students.
- Summarizing_Reading is the second option for Freshman students but it seems to be irrelevant for the other year categories.
- Debugging/Troubleshooting is important for Junior and Sophomore students but not for the other ones. 
- Direct_Answer_Generation is the second option for Graduate and Sophomore students but it seems to be irrelevant for the other year categories.

#### **General stats**

Columns are sorted in a different order to have a clearer view of information, then describe method is called to look for basic statistical indicators.

In [21]:
business_students_stats = business_students[['Pre_Semester_GPA', 'Post_Semester_GPA',
                                       'Traditional_Study_Hours', 'Weekly_GenAI_Hours',
                                       'Tool_Diversity', 'Perceived_AI_Dependency',
                                       'Anxiety_Level_During_Exams', 'Skill_Retention_Score']]
business_students_stats.describe()

,Pre_Semester_GPA,Post_Semester_GPA,Traditional_Study_Hours,Weekly_GenAI_Hours,Tool_Diversity,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Skill_Retention_Score
count,12538.000000,12538.000000,12538.000000,12538.000000,12538.000000,12538.000000,12538.000000,12538.000000
mean,3.141760,3.336085,11.205092,8.275898,2.740788,3.489312,4.250917,75.255889
std,0.480846,0.498114,5.136164,8.011074,1.180441,1.796877,2.130852,13.024232
min,1.183000,1.209000,1.000000,0.000000,1.000000,1.000000,1.000000,13.230000
25%,2.830000,3.010000,7.530000,2.430000,2.000000,2.000000,3.000000,66.562500
50%,3.206000,3.408000,11.180000,5.820000,3.000000,3.000000,4.000000,75.470000
75%,3.518000,3.735000,14.697500,11.460000,3.000000,5.000000,6.000000,84.317500
max,3.992000,4.000000,30.420000,40.000000,5.000000,10.000000,10.000000,100.000000


After analyzing Business students stats it can be given these preliminary conclusions:

- Traditional study hours remains as the main study method for the students.
- The mean of tool diversity is 2.74, which means that students aren't interested on exploring different types of AI tools.
- Perception of AI dependency by students is low and anxiety level too, so this variables cannot prove a negative impact of AI usage.
- When GPA mean and max on pre and post semester is analyzed it can be concluded that there's no real impact of AI on student performance, that's because GPA post is only 0.19 points higher than GPA pre semester, and the maximum grade remains almost the same.

#### **Burnout risk**

The column Burnout_Risk_Level represents an assessment applied to students to measure risk of having burnout after completing one semester.

A count is made to know that is the most frequent risk type.

In [22]:
business_burnout_risk = business_students['Burnout_Risk_Level'].value_counts().reset_index()
business_burnout_risk['Percentaje'] = round(business_burnout_risk['count'] / 12538 * 100, 2)
business_burnout_risk

,Burnout_Risk_Level,count,Percentaje
0,Medium,5294,42.22
1,Low,4193,33.44
2,High,3051,24.33


Medium level is the main and High the last like in Arts students but with a bit more and less in each one.

A count of students that have an specific risk level is made to know if there is a variation in the different years of study.

In [23]:
business_s_year_bnt = business_students.groupby(["Year_of_Study","Burnout_Risk_Level"]).count()
business_s_year_bnt = business_s_year_bnt[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
business_s_year_bnt = (
    business_s_year_bnt
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
business_s_year_bnt = business_s_year_bnt[["Year_of_Study","Burnout_Risk_Level","Student_ID"]]
business_s_year_bnt

,Year_of_Study,Burnout_Risk_Level,Student_ID
0,Freshman,Low,1209
1,Freshman,Medium,1129
2,Freshman,High,493
3,Graduate,Medium,797
4,Graduate,High,664
5,Graduate,Low,384
6,Junior,Medium,1223
7,Junior,Low,937
8,Junior,High,685
9,Senior,Medium,1096


As seen in the table Freshman is the only Year_of_Study category that has a different main risk level.

### Humanities students

A new Dataframe is created to select only students of Humanities

In [24]:
humanities_students = df.loc[df.Major_Category == "Humanities"]
humanities_students

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,100001,Humanities,Senior,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
16,100017,Humanities,Sophomore,2.931,2.08,Direct_Answer_Generation,Advanced,3,False,15.15,5,Strict_Ban,8,3.214,59.69,Medium
18,100019,Humanities,Junior,2.975,0.51,Summarizing_Reading,Beginner,1,False,10.84,1,Allowed_With_Citation,2,3.412,53.95,Medium
23,100024,Humanities,Senior,3.579,0.45,Ideation,Beginner,1,True,10.61,1,Allowed_With_Citation,8,3.760,55.33,Low
24,100025,Humanities,Freshman,2.364,28.72,Copywriting/Drafting,Beginner,1,False,8.24,8,Strict_Ban,7,2.588,63.38,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49972,149973,Humanities,Senior,2.540,8.05,Direct_Answer_Generation,Advanced,4,False,14.73,3,Allowed_With_Citation,4,3.029,100.00,High
49985,149986,Humanities,Sophomore,3.661,1.91,Copywriting/Drafting,Beginner,3,True,14.19,3,Actively_Encouraged,2,3.910,84.82,Low
49987,149988,Humanities,Junior,2.980,0.50,Direct_Answer_Generation,Intermediate,2,False,16.07,4,Allowed_With_Citation,2,3.568,86.06,Low
49991,149992,Humanities,Freshman,2.708,13.34,Copywriting/Drafting,Intermediate,2,False,11.65,4,Allowed_With_Citation,5,2.785,65.24,High


A count of students per year of study is made to know how many there are in each year.

In [25]:
humanities_students["Year_of_Study"].value_counts().sort_values()

Year_of_Study
Graduate     1520
Sophomore    1944
Senior       2155
Freshman     2174
Junior       2201
Name: count, dtype: int64

A count of students that have an specific use of AI is made to know what is the most frequent use in each year of study.

In [26]:
humanities_s_AI_use = humanities_students.groupby(["Year_of_Study","Primary_Use_Case"]).count()
humanities_s_AI_use = humanities_s_AI_use[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
humanities_s_AI_use = (
    humanities_s_AI_use
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
humanities_s_AI_use = humanities_s_AI_use[["Year_of_Study","Primary_Use_Case","Student_ID"]]
humanities_s_AI_use

,Year_of_Study,Primary_Use_Case,Student_ID
0,Freshman,Copywriting/Drafting,1111
1,Freshman,Ideation,283
2,Freshman,Summarizing_Reading,282
3,Freshman,Debugging/Troubleshooting,250
4,Freshman,Direct_Answer_Generation,248
5,Graduate,Copywriting/Drafting,814
6,Graduate,Ideation,184
7,Graduate,Debugging/Troubleshooting,181
8,Graduate,Summarizing_Reading,173
9,Graduate,Direct_Answer_Generation,168


This table shows that Copywriting/Drafting is the main use in students of every year category, but the other use cases differ in the different categories. 

With this info it can be given the following statements: 

- Ideation is the second option for Freshman, Graduate and Junior students.
- Summarizing_Reading is only important for Senior and Sophomore students.
- Debugging/Troubleshooting	seems to be irrelevant among all types of students. 
- Direct_Answer_Generation is the less important use for Freshman, Graduate and Junior students.

#### **General stats**

Columns are sorted in a different order to have a clearer view of information, then describe method is called to look for basic statistical indicators.

In [27]:
humanities_students_stats = humanities_students[['Pre_Semester_GPA', 'Post_Semester_GPA',
                                       'Traditional_Study_Hours', 'Weekly_GenAI_Hours',
                                       'Tool_Diversity', 'Perceived_AI_Dependency',
                                       'Anxiety_Level_During_Exams', 'Skill_Retention_Score']]
humanities_students_stats.describe()

,Pre_Semester_GPA,Post_Semester_GPA,Traditional_Study_Hours,Weekly_GenAI_Hours,Tool_Diversity,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Skill_Retention_Score
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,3.147850,3.345825,11.336752,6.770144,2.744347,3.243646,4.142485,75.253767
std,0.477493,0.492503,5.108085,6.586853,1.181252,1.644952,2.078108,12.955501
min,1.342000,1.190000,1.000000,0.000000,1.000000,1.000000,1.000000,14.880000
25%,2.833000,3.029000,7.770000,1.930000,2.000000,2.000000,3.000000,66.422500
50%,3.210000,3.411000,11.340000,4.720000,3.000000,3.000000,4.000000,75.540000
75%,3.520000,3.742000,14.860000,9.457500,3.000000,4.000000,6.000000,84.455000
max,3.997000,4.000000,30.710000,34.000000,5.000000,10.000000,10.000000,100.000000


After analyzing Humanities students stats it can be applied the same conclusions given to Business students, the variation is minimal.

#### **Burnout risk**

The column Burnout_Risk_Level represents an assessment applied to students to measure risk of having burnout after completing one semester.

A count is made to know that is the most frequent risk type.

In [28]:
humanities_burnout_risk = humanities_students['Burnout_Risk_Level'].value_counts().reset_index()
humanities_burnout_risk['Percentaje'] = round(humanities_burnout_risk['count'] / 9994 * 100, 2)
humanities_burnout_risk

,Burnout_Risk_Level,count,Percentaje
0,Medium,4315,43.18
1,Low,3606,36.08
2,High,2073,20.74


Risk levels remains almost the same as in other student categories

A count of students that have an specific risk level is made to know if there is a variation in the different years of study.

In [29]:
humanities_s_year_bnt = humanities_students.groupby(["Year_of_Study","Burnout_Risk_Level"]).count()
humanities_s_year_bnt = humanities_s_year_bnt[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
humanities_s_year_bnt = (
    humanities_s_year_bnt
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
humanities_s_year_bnt = humanities_s_year_bnt[["Year_of_Study","Burnout_Risk_Level","Student_ID"]]
humanities_s_year_bnt

,Year_of_Study,Burnout_Risk_Level,Student_ID
0,Freshman,Low,1000
1,Freshman,Medium,884
2,Freshman,High,290
3,Graduate,Medium,657
4,Graduate,High,512
5,Graduate,Low,351
6,Junior,Medium,971
7,Junior,Low,794
8,Junior,High,436
9,Senior,Medium,977


As seen in the table Freshman is the only Year_of_Study category that has a different main risk level.

### Medicine students

A new Dataframe is created to select only students of Medicine

In [ ]:
medicine_students = df.loc[df.Major_Category == "Medical"]
medicine_students

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
1,100002,Medical,Junior,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
11,100012,Medical,Junior,2.627,3.62,Direct_Answer_Generation,Intermediate,3,False,8.05,3,Actively_Encouraged,3,2.699,72.41,Medium
33,100034,Medical,Senior,3.148,5.56,Summarizing_Reading,Intermediate,5,False,9.89,2,Allowed_With_Citation,1,3.335,89.14,Medium
34,100035,Medical,Graduate,3.120,19.92,Summarizing_Reading,Beginner,2,True,1.00,7,Allowed_With_Citation,3,3.207,67.20,High
43,100044,Medical,Senior,3.048,4.67,Summarizing_Reading,Beginner,3,False,8.21,4,Strict_Ban,4,3.277,79.79,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49940,149941,Medical,Freshman,3.545,15.65,Summarizing_Reading,Beginner,1,True,4.23,3,Allowed_With_Citation,2,3.698,55.71,High
49949,149950,Medical,Graduate,3.435,1.48,Copywriting/Drafting,Beginner,2,False,9.42,1,Actively_Encouraged,6,3.660,57.03,Medium
49966,149967,Medical,Junior,2.921,22.68,Summarizing_Reading,Advanced,5,True,4.56,7,Allowed_With_Citation,4,3.150,71.18,High
49971,149972,Medical,Freshman,2.679,14.44,Summarizing_Reading,Beginner,3,False,19.77,5,Actively_Encouraged,6,2.936,65.28,Medium


A count of students per year of study is made to know how many there are in each year.

In [31]:
medicine_students["Year_of_Study"].value_counts().sort_values()

Year_of_Study
Graduate      953
Sophomore    1308
Senior       1355
Junior       1396
Freshman     1464
Name: count, dtype: int64

A count of students that have an specific use of AI is made to know what is the most frequent use in each year of study.

In [32]:
medicine_s_AI_use = medicine_students.groupby(["Year_of_Study","Primary_Use_Case"]).count()
medicine_s_AI_use = medicine_s_AI_use[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
medicine_s_AI_use = (
    medicine_s_AI_use
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
medicine_s_AI_use = medicine_s_AI_use[["Year_of_Study","Primary_Use_Case","Student_ID"]]
medicine_s_AI_use

,Year_of_Study,Primary_Use_Case,Student_ID
0,Freshman,Summarizing_Reading,713
1,Freshman,Ideation,193
2,Freshman,Debugging/Troubleshooting,192
3,Freshman,Copywriting/Drafting,187
4,Freshman,Direct_Answer_Generation,179
5,Graduate,Summarizing_Reading,441
6,Graduate,Direct_Answer_Generation,149
7,Graduate,Ideation,133
8,Graduate,Copywriting/Drafting,116
9,Graduate,Debugging/Troubleshooting,114


This table shows that Summarizing_Reading is the main use in students of every year category, but the other use cases differ in the different categories. 

With this info it can be given the following statements: 

- Ideation is an important use case for Freshman but irrelevant for other year categories.
- Debugging/Troubleshooting	is the second option for Junior and Senior students and the third one for Freshman and Sophomore.
- Direct_Answer_Generation is the second option for Graduate and Sophomore but irrelevant for other year categories.
- Copywriting/Drafting seems to be irrelevant for all medicine students. 

#### **General stats**

Columns are sorted in a different order to have a clearer view of information, then describe method is called to look for basic statistical indicators.

In [ ]:
medicine_students_stats = medicine_students[['Pre_Semester_GPA', 'Post_Semester_GPA',
                                       'Traditional_Study_Hours', 'Weekly_GenAI_Hours',
                                       'Tool_Diversity', 'Perceived_AI_Dependency',
                                       'Anxiety_Level_During_Exams', 'Skill_Retention_Score']]
medicine_students_stats.describe()

,Pre_Semester_GPA,Post_Semester_GPA,Traditional_Study_Hours,Weekly_GenAI_Hours,Tool_Diversity,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Skill_Retention_Score
count,6476.000000,6476.000000,6476.000000,6476.000000,6476.000000,6476.000000,6476.000000,6476.000000
mean,3.151769,3.353168,11.365145,7.546728,2.767295,3.399629,4.227610,75.484605
std,0.477011,0.494697,5.206199,7.410775,1.189420,1.734312,2.122158,13.143772
min,1.317000,1.226000,1.000000,0.000000,1.000000,1.000000,1.000000,21.250000
25%,2.846750,3.031750,7.750000,2.150000,2.000000,2.000000,3.000000,66.547500
50%,3.211500,3.424000,11.320000,5.240000,3.000000,3.000000,4.000000,75.645000
75%,3.531000,3.747000,14.840000,10.550000,4.000000,4.000000,6.000000,84.742500
max,3.998000,4.000000,35.860000,38.000000,5.000000,10.000000,10.000000,100.000000


After analyzing Medicine students stats it can be applied the same conclusions given to Humanities students, the variation is minimal.

#### **Burnout risk**

The column Burnout_Risk_Level represents an assessment applied to students to measure risk of having burnout after completing one semester.

A count is made to know that is the most frequent risk type.

In [36]:
medicine_burnout_risk = medicine_students['Burnout_Risk_Level'].value_counts().reset_index()
medicine_burnout_risk['Percentaje'] = round(medicine_burnout_risk['count'] / 6476 * 100, 2)
medicine_burnout_risk

,Burnout_Risk_Level,count,Percentaje
0,Medium,2733,42.20
1,Low,2243,34.64
2,High,1500,23.16


Risk levels remains almost the same as in other student categories

A count of students that have an specific risk level is made to know if there is a variation in the different years of study.

In [37]:
medicine_s_year_bnt = medicine_students.groupby(["Year_of_Study","Burnout_Risk_Level"]).count()
medicine_s_year_bnt = medicine_s_year_bnt[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
medicine_s_year_bnt = (
    medicine_s_year_bnt
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
medicine_s_year_bnt = medicine_s_year_bnt[["Year_of_Study","Burnout_Risk_Level","Student_ID"]]
medicine_s_year_bnt

,Year_of_Study,Burnout_Risk_Level,Student_ID
0,Freshman,Low,641
1,Freshman,Medium,584
2,Freshman,High,239
3,Graduate,Medium,422
4,Graduate,High,352
5,Graduate,Low,179
6,Junior,Medium,593
7,Junior,Low,496
8,Junior,High,307
9,Senior,Medium,603


As seen in the table Freshman is the only Year_of_Study category that has a different main risk level.

### STEM Students

A new Dataframe is created to select only students of Science, Technology, Engineering, and Mathematics

In [38]:
stem_students = df.loc[df.Major_Category == "STEM"]
stem_students

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
4,100005,STEM,Sophomore,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium
5,100006,STEM,Junior,3.449,6.50,Debugging/Troubleshooting,Beginner,1,False,14.19,4,Allowed_With_Citation,5,3.666,65.92,High
6,100007,STEM,Freshman,3.622,31.41,Summarizing_Reading,Advanced,5,True,13.11,8,Allowed_With_Citation,7,4.000,67.97,Medium
10,100011,STEM,Graduate,3.407,5.16,Summarizing_Reading,Advanced,3,False,11.43,5,Allowed_With_Citation,3,3.617,95.42,High
13,100014,STEM,Junior,3.225,8.95,Direct_Answer_Generation,Beginner,1,False,25.30,2,Allowed_With_Citation,3,3.935,64.47,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49980,149981,STEM,Graduate,3.419,25.47,Summarizing_Reading,Advanced,4,True,14.24,6,Allowed_With_Citation,7,3.879,66.16,High
49981,149982,STEM,Graduate,2.256,10.89,Copywriting/Drafting,Advanced,4,False,10.42,1,Allowed_With_Citation,1,2.450,84.88,Low
49988,149989,STEM,Sophomore,2.058,3.32,Debugging/Troubleshooting,Beginner,1,False,6.34,4,Actively_Encouraged,4,2.147,68.75,Low
49993,149994,STEM,Sophomore,3.588,25.64,Debugging/Troubleshooting,Intermediate,3,True,4.62,5,Strict_Ban,10,3.841,58.44,High


A count of students per year of study is made to know how many there are in each year.

In [39]:
stem_students["Year_of_Study"].value_counts().sort_values()

Year_of_Study
Graduate     2268
Sophomore    2959
Senior       3273
Junior       3279
Freshman     3280
Name: count, dtype: int64

A count of students that have an specific use of AI is made to know what is the most frequent use in each year of study.

In [40]:
stem_s_AI_use = stem_students.groupby(["Year_of_Study","Primary_Use_Case"]).count()
stem_s_AI_use = stem_s_AI_use[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
stem_s_AI_use = (
    stem_s_AI_use
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
stem_s_AI_use = stem_s_AI_use[["Year_of_Study","Primary_Use_Case","Student_ID"]]
stem_s_AI_use

,Year_of_Study,Primary_Use_Case,Student_ID
0,Freshman,Debugging/Troubleshooting,1693
1,Freshman,Summarizing_Reading,413
2,Freshman,Direct_Answer_Generation,408
3,Freshman,Copywriting/Drafting,400
4,Freshman,Ideation,366
5,Graduate,Debugging/Troubleshooting,1155
6,Graduate,Direct_Answer_Generation,306
7,Graduate,Summarizing_Reading,281
8,Graduate,Ideation,274
9,Graduate,Copywriting/Drafting,252


This table shows that Debugging/Troubleshooting is the main use in students of every year category, but the other use cases differ in the different categories. 

With this info it can be given the following statements: 

- Summarizing_Reading is the second option for Freshman but irrelevant for other year categories.
- Copywriting/Drafting is important for Sophomore students but not for other year categories.
- Direct_Answer_Generation is the second option for Junior, Graduate and Senior.
- Ideation seems to be irrelevant for all STEM students. 

#### **General stats**

Columns are sorted in a different order to have a clearer view of information, then describe method is called to look for basic statistical indicators.

In [41]:
stem_students_stats = stem_students[['Pre_Semester_GPA', 'Post_Semester_GPA',
                                       'Traditional_Study_Hours', 'Weekly_GenAI_Hours',
                                       'Tool_Diversity', 'Perceived_AI_Dependency',
                                       'Anxiety_Level_During_Exams', 'Skill_Retention_Score']]
stem_students_stats.describe()

,Pre_Semester_GPA,Post_Semester_GPA,Traditional_Study_Hours,Weekly_GenAI_Hours,Tool_Diversity,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Skill_Retention_Score
count,15059.000000,15059.000000,15059.000000,15059.000000,15059.000000,15059.000000,15059.000000,15059.000000
mean,3.145852,3.363154,11.004719,10.488609,2.915864,3.793811,4.428979,76.799725
std,0.478498,0.495586,5.182034,9.733261,1.188737,1.983810,2.215013,13.817070
min,1.275000,1.000000,1.000000,0.000000,1.000000,1.000000,1.000000,10.780000
25%,2.833000,3.034000,7.310000,3.120000,2.000000,2.000000,3.000000,67.440000
50%,3.210000,3.442000,10.980000,7.400000,3.000000,4.000000,4.000000,77.160000
75%,3.518000,3.767000,14.515000,14.910000,4.000000,5.000000,6.000000,86.775000
max,3.991000,4.000000,31.210000,40.000000,5.000000,10.000000,10.000000,100.000000


After analyzing Medicine students stats it can be applied the same conclusions given to other student categories, the variation is minimal.

#### **Burnout risk**

The column Burnout_Risk_Level represents an assessment applied to students to measure risk of having burnout after completing one semester.

A count is made to know that is the most frequent risk type.

In [43]:
stem_burnout_risk = stem_students['Burnout_Risk_Level'].value_counts().reset_index()
stem_burnout_risk['Percentaje'] = round(stem_burnout_risk['count'] / 15059 * 100, 2)
stem_burnout_risk

,Burnout_Risk_Level,count,Percentaje
0,Medium,6229,41.36
1,High,4517,30.00
2,Low,4313,28.64


Risk levels remains almost the same as in other student categories

A count of students that have an specific risk level is made to know if there is a variation in the different years of study.

In [44]:
stem_s_year_bnt = stem_students.groupby(["Year_of_Study","Burnout_Risk_Level"]).count()
stem_s_year_bnt = stem_s_year_bnt[["Student_ID"]]
# A sort is made based on Year_of_Study and the count of Student_ID
stem_s_year_bnt = (
    stem_s_year_bnt
    .reset_index() # Indexes are reset to convert them into columns
    .sort_values(["Year_of_Study", "Student_ID"], ascending=[True, False])
    .reset_index(drop=True)
)
# Filter of columns to avoid showing index columns
stem_s_year_bnt = stem_s_year_bnt[["Year_of_Study","Burnout_Risk_Level","Student_ID"]]
stem_s_year_bnt

,Year_of_Study,Burnout_Risk_Level,Student_ID
0,Freshman,Medium,1309
1,Freshman,Low,1220
2,Freshman,High,751
3,Graduate,Medium,940
4,Graduate,High,913
5,Graduate,Low,415
6,Junior,Medium,1398
7,Junior,High,974
8,Junior,Low,907
9,Senior,Medium,1392


There's no variation, all STEM students have a Medium level.

## Conclusions

After analyzing all statistics it is concluded that there's no real impact of AI usage on college students, this is because there's no significant improve on GPA after finishing one semester, the maximum grade achieved by some students remains the same befor and after finishing one semester. 

The anxiety level and perceived AI dependency is low for all students, the skill retention is on the average, therefore, this leads to conclude that the use of AI tools don't have a negative impact on mental health at least in the short term. 

Impact on mental health cannot be measured with certainty because the mean of traditional study hours is higher than hours spent with AI assistance, so students see AI tools as a complement and another tool for autonomous study.

AI subscription was not considered in this analysis due to lack of information about type of paid plans and amount of money spent on subscriptions, therefore, a relation between paid AI subscription and study improvement cannot be made.

Regarding AI usage, it was found the following: 

- Copywriting/Drafting is the main use case for Arts and Humanities students.
- Ideation is the main use case for Business students.
- Summarizing_Reading is the main use case for Medicine students.
- Debugging/Troubleshooting is the main use case for STEM students.
- Direct_Answer_Generation is not relevant for most of students. 